libs-scala/oracle-testing/src/main/scala/testing/oracle/OracleAround.scala

```diff
@@ -5,86 +5,127 @@ package com.daml.testing.oracle
 
 import com.daml.ports._
 import java.sql._
-import scala.util.{Random, Using}
 
-private[daml] final case class User(name: String, pwd: String)
+import org.slf4j.LoggerFactory
 
-trait OracleAround {
-  @volatile
-  private var systemUser: String = _
-  @volatile
-  private var systemPwd: String = _
-  @volatile
-  private var port: Port = _
+import scala.annotation.tailrec
+import scala.util.{Failure, Random, Success, Try, Using}
 
-  def oraclePort: Port = port
+object OracleAround {
 
-  def oracleJdbcUrl: String = s"jdbc:oracle:thin:@localhost:$oraclePort/ORCLPDB1"
+  case class RichOracleUser(
+      oracleUser: OracleUser,
+      jdbcUrlWithoutCredentials: String,
+      jdbcUrl: String,
+      drop: () => Unit,
+  ) {
 
-  protected def connectToOracle(): Unit = {
-    systemUser = sys.env("ORACLE_USERNAME")
-    systemPwd = sys.env("ORACLE_PWD")
-    port = Port(sys.env("ORACLE_PORT").toInt)
+    /** In CI we re-use the same Oracle instance for testing, so non-colliding DB-lock-ids need to be assigned
+      *
+      * @return A positive integer, which defines a unique / mutually exclusive range of usable lock ids: [seed, seed + 10)
+      */
+    def lockIdSeed: Int = {
+      assert(oracleUser.id > 0, "Lock ID seeding is not supported, cannot ensure unique lock-ids")
+      assert(oracleUser.id < 10 * 1000 * 1000)
+      oracleUser.id * 10
+    }
   }
 
-  protected def createNewRandomUser(): User = {
-    // See https://docs.oracle.com/cd/B19306_01/server.102/b14200/sql_elements008.htm#i27570
-    // for name restrictions.
-    val u = "u" + Random.alphanumeric.take(29).mkString("")
-    createNewUser(u.toUpperCase)
+  case class OracleUser(name: String, id: Int) {
+    val pwd: String = "hunter2"
   }
 
-  protected def checkUserExists(con: Connection, name: String): Boolean = {
-    Using(con.createStatement()) { stmt =>
-      val res = stmt.executeQuery(
-        s"""SELECT count(*) AS user_count FROM all_users WHERE username='${name.toUpperCase}'"""
-      )
-      val count = if (res.next()) {
-        res.getInt("user_count")
-      } else 0
-      count > 0
-    }.get
+  private val logger = LoggerFactory.getLogger(getClass)
+
+  def createNewUniqueRandomUser(): RichOracleUser = createRichOracleUser { stmt =>
```

Comment:
this is the usual use case: having mutually exclusive lockIdSeed and unique user